Support original PDF

daipom opened the pull request:
Support original PDF

`:pdf` markup language already exists, but we couldn't use it to use original PDF.
This fix allows us to use `:pdf` markup language to use original PDF.

Points:

* Add `--readme-markup-language` to manage README format.
  * Use `:markdown` for `:pdf`.
  * Otherwise, use the same as `--markup-language` by default.
  * Backward compatible.
* The source file is `pdf/{id}-{base-name}.pdf`.
  * Copy the PDF file specified by the `--pdf` option at initialization.
* Skip `pdf` task to avoid unexpected PDF rebuild.

How to use:

1. `rabbit-slide new --id=foo --base_name=bar --markup-language=pdf --pdf=/path/to/pdf_file ...`
2. Write `README.md`.
3. Edit `config.yaml` if necessary.
4. You can run `rake` to show it or `rake publish` to publish it now.

daipom commented on lib/rabbit/command/rabbit-slide.rb:
```diff
@@ -772,7 +782,12 @@ def generate_slide
-        "#{@data.slide_conf.base_name}.#{slide_source_extension}"
+        case @data.author_conf.markup_language
+        when :pdf
+          File.join("pdf", "#{@data.slide_conf.id}-#{@data.slide_conf.base_name}.pdf")
```
Note: It would be better to unify this with the following, but I can't think of a good way to do it.

https://github.com/rabbit-shocker/rabbit/blob/01885b247a230b4b122831cd48d5d8935b759879/lib/rabbit/task/slide.rb#L203-L205

daipom replied:
Thanks! Nice idea! Fixed!

kou commented on lib/rabbit/command/rabbit-slide.rb:
```diff
@@ -705,6 +707,7 @@ def generate_readme
 
       def readme_content
         markup_language = @data.markup_language
+        markup_language = :markdown if markup_language == :pdf
```
Hmm. Let's add `--readme-markup-language` instead of reusing `--markup-language`. (We can use `--markup-language` value as the default value of `--readme-markup-language`.)

daipom replied:
Maybe we should consider this after #183?

daipom replied:
DONE.

kou commented on lib/rabbit/task/slide.rb:
```diff
@@ -145,6 +145,10 @@ def define_gem_validate_task
+          if @slide.author.markup_language == :pdf
+            puts "Skipped pdf task because markup_language is PDF."
+            next
+          end
```
Does this work?

```diff
diff --git a/lib/rabbit/task/slide.rb b/lib/rabbit/task/slide.rb
index 2e320532..69b0cac3 100644
--- a/lib/rabbit/task/slide.rb
+++ b/lib/rabbit/task/slide.rb
@@ -144,14 +144,19 @@ module Rabbit
       end
 
       def define_pdf_task
-        file pdf_path => [options_path, *(spec.files - [pdf_path])] do
-          mkdir_p(@pdf_dir)
-          rabbit("--print",
-                 "--output-filename", pdf_path)
-        end
+        if @slide.author.markup_language == :pdf
+          # Does nothing. Not list in "rake -T" but "rake pdf" is accepted..
+          task :pdf
+        else
+          file pdf_path => [options_path, *(spec.files - [pdf_path])] do
+            mkdir_p(@pdf_dir)
+            rabbit("--print",
+                   "--output-filename", pdf_path)
+          end
 
-        desc(_("Generate PDF: %{pdf_path}") % {:pdf_path => pdf_path})
-        task :pdf => pdf_path
+          desc(_("Generate PDF: %{pdf_path}") % {:pdf_path => pdf_path})
+          task :pdf => pdf_path
+        end
       end
 
       def define_publish_task
```

daipom replied:
Thanks! This works!

kou reviewed (commented):
It seems that we should move `markup_language` to `SlideConfiguration` from `AuthorConfiguraiton` before this.

We may use different markup languages over slides.

daipom commented:
> > It seems that we should move `markup_language` to `SlideConfiguration` from `AuthorConfiguraiton` before this.
> 
> I’ve fixed everything else related to this. I’ll make a separate PR for this.

* #183

daipom committed:
Support GUI for `--pdf`

daipom commented:
Applied #183.

TODO

* Add `--readme-markup-language`.
* I18n.

daipom committed:
Support original PDF

`:pdf` markup language already exists, but we couldn't use it to
use original PDF.
This fix allows us to use `:pdf` markup language to use original
PDF.

Points:

* Add `--readme-markup-language` to manage README format.
  * Use `:markdown` for `:pdf`.
  * Otherwise, use the same as `--markup-language` by default.
  * Backward compatible.
* The source file is `pdf/{id}-{base-name}.pdf`.
  * Copy the PDF file specified by the `--pdf` option at initialization.
* Skip `pdf` task to avoid unexpected PDF rebuild.

How to use:

1. `rabbit-slide new --id=foo --base_name=bar --markup-language=pdf --pdf=/path/to/pdf_file ...`
2. Write `README.md`.
3. Edit `config.yaml` if necessary.
4. You can run `rake` to show it or `rake publish` to publish it now.

Co-authored-by: Sutou Kouhei <kou@clear-code.com>

daipom commented:
> * Add `--readme-markup-language`.

DONE.
